polyply/tests/test_apply_modifications.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2024 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Test that termini modifications are applied
+"""
+import logging
+from contextlib import nullcontext as does_not_raise
+from .example_fixtures import example_meta_molecule
+import pytest
+import vermouth.forcefield
+import vermouth.ffinput
+from polyply.src.meta_molecule import MetaMolecule
+from polyply.src.apply_modifications import _patch_protein_termini, apply_mod, ApplyModifications
+from polyply import TEST_DATA
+import polyply.src.ff_parser_sub
+import networkx as nx
+from vermouth.molecule import Interaction
+
+def test_annotate_protein(example_meta_molecule):
+    """"
+    test that a protein sequence gets annotated correctly
+    """
```

Comment:
please also add a for the case where only one is given and annotated to both ends